src/NHibernate/Engine/CascadeStyle.cs

```diff
@@ -78,7 +78,7 @@ public virtual bool HasOrphanDelete
 
 		#region  Static helper methods
 
-		private static readonly Dictionary<string, CascadeStyle> Styles = new Dictionary<string, CascadeStyle>();
+		private static readonly Dictionary<string, CascadeStyle> Styles = new Dictionary<string, CascadeStyle>(StringComparer.OrdinalIgnoreCase);
```

Comment:
This looks to me as an undue change. Cascade styles are not case insensitive, even if currently we would not have names conflict by going insensitive.

Reply:
Are you sure? Change is based on the following line:

https://github.com/nhibernate/nhibernate-core/blob/4e84dbc4da655c9f63923282fca3e4a511234638/src/NHibernate/Mapping/Property.cs#L114

Reply:
Well, so the internal usage of this public class is case insensitive. It is then unlikely it could be a breaking change for anyone to render the class case insensitive, but not strictly impossible either.

Well, keep your change in place.